L/libdecor/build_tarballs.jl

```diff
@@ -0,0 +1,57 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder
+
+name = "libdecor"
+version = v"0.2.2"
+
+# Collection of sources required to build libdecor
+sources = [
+    ArchiveSource("https://gitlab.freedesktop.org/libdecor/libdecor/-/archive/$(version)/libdecor-$(version).tar.gz",
+                  "40a1d8be07d8b1f66e8fb98a1f4a84549ca6bf992407198a5055952be80a8525"),
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir/libdecor-*/
+
+mkdir build && cd build
+meson setup .. \
+    --cross-file="${MESON_TARGET_TOOLCHAIN}" \
+    --prefix=${prefix} \
+    --buildtype=release \
+    -Ddemo=false \
+    -Dgtk=disabled
+ninja -j${nproc}
+ninja install
+"""
+
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+platforms = filter(p -> arch(p) != "armv6l" && Sys.islinux(p), supported_platforms())
+
+# The products that we will ensure are always built
+products = [
+    LibraryProduct("libdecor-0", :libdecor),
+]
+
+# Dependencies that must be installed before this package can be built
+dependencies = [
+    Dependency("xkbcommon_jll"; platforms),
+    Dependency("Dbus_jll"; platforms),
+    Dependency("Libglvnd_jll"; platforms),
+    Dependency("Pango_jll"; platforms),
+    Dependency("Wayland_jll"; platforms),
+    HostBuildDependency("Wayland_jll"; platforms),
+    BuildDependency("Wayland_protocols_jll"; platforms),
+    BuildDependency("Xorg_xorgproto_jll"; platforms),
```

Comment:
Any particular reason for filtering for all platforms anyway?

Reply:
Maybe no good reason — I was assuming everything should only try to load on a Wayland-compatible system, so this was my guess as to how to do that. If this isn't the right way to do it, I'm happy to learn what's the recommended practice.

Reply:
With `platforms=platforms` you're making a dependency be loaded only for the listed platforms, but when these are just _all_ the platforms I miss the point of filtering in the first place, since you aren't filtering anything.

Reply:
I guess I wasn't sure how the list of build platforms interacts with the generated Julia package's `using`s. If the "implicit" platform list is sufficient, I'll just get rid of these.